### PR TITLE
[New Egg] SourceCoop

### PR DESCRIPTION
**Pull request by lilkingjr1**
# Description

Adds a new Egg for [SourceCoop](https://github.com/ampreeT/SourceCoop): A cooperative server-side-only mod for multi-player Source Engine games (like Black Mesa), that enables players to play together on single-player campaigns or maps. This Egg currently only supports Black Mesa (the mod's primary game of focus) and it's official maps, but this may be expanded to more games in the future via additional Eggs. Please see the README for additional information.

Also, this PR adjusts the existing Black Mesa Egg's README to fix typos and make it more clear of the differences between that Egg and this one.

Sort-of fulfills @shodanx2's Egg request as a comment in this issue: https://github.com/pelican-eggs/eggs/issues/1856

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
6. [x] The egg was exported from the panel

### CODE REVIEW

File: sourcecoop/egg-source-coop.json [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
+    "meta": {
+        "version": "PTDL_v2",
+        "update_url": null
+    },
+    "exported_at": "2024-05-14T22:54:17-07:00",
+    "name": "SourceCoop",
+    "author": "red_thirten@yahoo.com",
+    "description": "SourceCoop is a cooperative server-side-only mod for multi-player Source Engine games (like Black Mesa), that enables players to play together on single-player campaigns or maps. This Egg currently supports official Black Mesa maps only, but can be easily modified to support more (see Egg README for details).",
+    "features": null,
```
Comment (QuintenQVD0) [inline sourcecoop/egg-source-coop.json:11]: no steam disk features?

File: sourcecoop/egg-source-coop.json [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    "docker_images": {
+        "ghcr.io\/parkervcp\/steamcmd:debian": "ghcr.io\/parkervcp\/steamcmd:debian"
```
Comment (QuintenQVD0) [inline sourcecoop/egg-source-coop.json:13]: This is a source game so should it not use the source image? (not that their diffrend at all :) )

File: sourcecoop/egg-source-coop.json [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
...endlyfire 0\r\n\r\n\/\/ setting 1 disables sprinting\r\nsv_always_run 0\r\n\r\n\/\/ disables map timelimit\r\nmp_timelimit 0\r\n\r\n\/\/ changes maps in empty server\r\nmp_mapcycle_empty_timeout_seconds 3600\r\n\r\n\/\/ Uncomment to loop back to the first map after finishing a campaign\r\n\/\/ Leave commented to use nextmap (mapcycle) instead\r\nsourcecoop_homemap \"bm_c0a0a\"\r\n\r\n\/\/ Enables pre-caching of map data (eg. models, etc.)\r\n\/\/ Disabling this may resolve crashing\/hanging on certain maps (namely bm_c4a4a), but may cause performance issues on other maps.\r\nmod_forcetouchdata 1\r\n\r\nrcon_password \"\" \/\/ disable rcon\r\nsv_tags \"custom, coop, sourcecoop\"\r\nsv_cheats 0\r\nsv_alltalk 1\r\nEOF\r\n\r\n## Check for successful installation.\r\ncd $HOME\/bms\/addons\r\nif [ -d \"metamod\" ] && [ -d \"sourcemod\" ] && [ -f \"sourcemod\/plugins\/srccoop.smx\" ]; then\r\n    echo -e \"\\nSourceCoop Dedicated Server successfully installed!\\n\"\r\nelse\r\n    echo -e \"\\n\\n[Error] One or more SourceCoop dependancies failed to install!\"\r\n    echo -e \"\\tTry checking your [Advanced] startup settings and\/or reinstalling the server again.\\n\"\r\n    exit 1\r\nfi",
```
Comment (QuintenQVD0) [inline sourcecoop/egg-source-coop.json:25]: Please use at least the full steamcmd install line if ever branches or logins want to be done that then their is only the need for just extra variables

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: sourcecoop/egg-source-coop.json (hunk 1 above)
Comment (lilkingjr1) [reply sourcecoop/egg-source-coop.json:11]: Ahh, sorry, forgot about these hidden features. I'll add that tomorrow. I wonder if Pelican will include them in the UI?

File: sourcecoop/egg-source-coop.json (hunk 3 above)
Comment (lilkingjr1) [reply sourcecoop/egg-source-coop.json:25]: That's fair; I'll add this in tomorrow as well :)

File: sourcecoop/egg-source-coop.json (hunk 1 above)
Comment (QuintenQVD0) [reply sourcecoop/egg-source-coop.json:11]: I hope so, I have asked them to add them.

File: sourcecoop/egg-source-coop.json (hunk 2 above)
Comment (QuintenQVD0) [reply sourcecoop/egg-source-coop.json:13]: no its fine, leave it at this as it is barely used.

File: sourcecoop/egg-source-coop.json (hunk 3 above)
Comment (lilkingjr1) [reply sourcecoop/egg-source-coop.json]: Done. Double check my syntax though if you don't mind 👍

File: sourcecoop/egg-source-coop.json (hunk 1 above)
Comment (lilkingjr1) [reply sourcecoop/egg-source-coop.json]: Added 👍

**Review by QuintenQVD0 — APPROVED** [review]
Looks all good to me

**Pull request merged**